src/foam/lib/query/AfterGtParser.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * @license
+ * Copyright 2018 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.lib.query;
+
+import foam.lib.parse.Literal;
+import foam.lib.parse.PStream;
+import foam.lib.parse.Parser;
+import foam.lib.parse.ParserContext;
+import foam.lib.parse.Seq1;
+
+public class AfterGtParser extends foam.lib.parse.ProxyParser {
+
+  public AfterGtParser(Parser valueParser) {
+    setDelegate(new Seq1(1,
+                         new Literal(">"),
+                         valueParser
+                        ));
+  }
+
+  @Override
+  public PStream parse(PStream ps, ParserContext x) {
+    ps = super.parse ( ps, x );
+    if ( ps == null ) return ps;
+
+    foam.mlang.predicate.Binary predicate = new foam.mlang.predicate.Gt ();
+    predicate.setArg1 ( ( foam.mlang.Expr ) x.get ( "arg1" ) );
+
+    if ( ps.value () instanceof Object[] )
+      predicate.setArg2 ( ( foam.mlang.Expr ) ( ( Object[] ) ps.value () )[0] );

```

Comment:
The only case that we will get Object[] is when we parse date. For each date parsed, we generate two date.(beginning and the end date, since the date contain milli seconds ).
The first date is the concrete date and the second date is the complement of the first.(Example d=2000/1; d1=2000/1/1 :00:00:00:000; d2=2000/1/31:24:59:59:999.
The second date in used only in the equal or in a partial date (like year).

Reply:
But this is the AfterGtParser, so it doesn't need to take an array of values.

 If you look at how the example I shared was structured, there's no way for the AfterParser to parse a date range, as it only includes a single date as its value.  This results in less fixing up of values in the parse() method, because we can only parse a single date for before/after situations.